src/GitHub.Exports/Models/DiffUtilities.cs

```diff
@@ -0,0 +1,124 @@
+                {
+                    var headerMatch = ChunkHeaderRegex.Match(line);
+
+                    if (headerMatch.Success)
+                    {
+                        if (chunk != null)
+                        {
+                            yield return chunk;
+                        }
+
+                        chunk = new DiffChunk
+                        {
+                            OldLineNumber = oldLine = int.Parse(headerMatch.Groups[1].Value),
+                            NewLineNumber = newLine = int.Parse(headerMatch.Groups[2].Value),
+                            DiffLine = diffLine,
+                        };
+                    }
+                    else if (chunk != null)
+                    {
+                        var type = GetLineChange(line[0]);
+                        if (type == DiffChangeType.Control)
+                        {
+                            // This might contain info about previous line (e.g. "\ No newline at end of file").
+                            continue;
+                        }
+
+                        chunk.Lines.Add(new DiffLine
+                        {
+                            Type = type,
+                            OldLineNumber = type != DiffChangeType.Add ? oldLine : -1,
+                            NewLineNumber = type != DiffChangeType.Delete ? newLine : -1,
+                            DiffLineNumber = diffLine,
+                            Content = line,
+                        });
+
+                        switch (type)
+                        {
+                            case DiffChangeType.None:
+                                ++oldLine;
+                                ++newLine;
+                                break;
+                            case DiffChangeType.Delete:
+                                ++oldLine;
+                                break;
+                            case DiffChangeType.Add:
+                                ++newLine;
+                                break;
+                        }
+                    }
+
+                    ++diffLine;
+                }
+
+                if (chunk != null)
+                {
+                    yield return chunk;
+                }
+            }
+        }
+
+        public static DiffLine Match(IEnumerable<DiffChunk> diff, IList<DiffLine> target)
+        {
+            int j = 0;
+
+            if (target.Count == 0)
+            {
+                return null; // no lines to match
+            }
+
+            foreach (var source in diff)
+            {
+                for (var i = source.Lines.Count - 1; i >= 0; --i)
+                {
+                    if (source.Lines[i].Content == target[j].Content)
+                    {
+                        if (++j == target.Count) return source.Lines[i + j - 1];
+                    }
+                    else
+                    {
+                        j = 0;
```

Comment:
Actually, one really simple thing we could do is treat `/` as white space when matching lines. That way a developer could `//` out lines and not immediately loose their inline comments.